### PR TITLE
add lock in undo log rollback

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/undo_log/undo_log.c [hunk 1]
```diff
@@ -286,6 +286,8 @@ void UndoLog_Rollback
 
 	if(count == 0) return;
 
+	QueryCtx_LockForCommit();
```
Comment (swilly22) [inline src/undo_log/undo_log.c:289]: // acquire lock before making any changes to the graph

File: src/undo_log/undo_log.c [hunk 2]
```diff
@@ -318,6 +320,8 @@ void UndoLog_Rollback
 		}
  	}
 
+	QueryCtx_ForceUnlockCommit();
```
Comment (swilly22) [inline src/undo_log/undo_log.c:323]: // force unlock, as we're not executing within the context of an operation
// assumption: no operations should be executing at this point

File: src/undo_log/undo_log.c [hunk 3]
```diff
@@ -286,6 +286,8 @@ void UndoLog_Rollback
 
 	if(count == 0) return;
 
+	QueryCtx_LockForCommit();
+
```
Comment (swilly22) [inline src/undo_log/undo_log.c:290]: consider computing the undo ranges (lines 296-301) outside of the locked block

**Review by swilly22 — CHANGES_REQUESTED** [review]
An important addition!
please see comments, let me suggest adding a test which timeouts on a query that have a short write phase followed by a long read phase the timeout is intended to occur on the READ part of the query.

File: src/undo_log/undo_log.c (hunk 3 above)
Comment (AviAvni) [reply src/undo_log/undo_log.c:291]: I don't feel like this worth it it is just decrementing a counter in a loop

**Review by swilly22 — COMMENTED** [review]
Please add a test, I think we want to make sure this mechanism works for a query such as:
```
CREATE() SHORT WRITE
LOOOONG READ period, experiance timeout at this point
```

**Review by swilly22 — APPROVED** [review]

**Pull request merged**